instrumentation/github.com/gocql/gocql/gocql_test.go

```diff
@@ -0,0 +1,526 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gocql
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/api/standard"
+
+	"go.opentelemetry.io/otel/sdk/metric/controller/push"
+	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
+
+	"github.com/gocql/gocql"
+	"github.com/stretchr/testify/assert"
+
+	mocktracer "go.opentelemetry.io/contrib/internal/trace"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/metric"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregation"
```

Comment:
We'll probably want to avoid this testing pattern where we depend on the default SDK. It will bring in a lot of dependencies for any user of this instrumentation (if they don't use the instrumentation (no SDK registered) it will still keep a copy of the SDK in the build because of this). Given the separation from the main repo, we cannot use the internal test structures like this https://github.com/open-telemetry/opentelemetry-go/blob/master/api/global/internal/meter_test.go#L28. We'll likely need to readdress when https://github.com/open-telemetry/opentelemetry-go/issues/872 is resolved.

This is a common issues we recently identified in other parts of the OTel API, so not uncommon.

Reply:
Tracking with https://github.com/open-telemetry/opentelemetry-go-contrib/issues/161